nile-backend/src/main/java/com/tp/Nile/controllers/SpecificationController.java

```diff
@@ -0,0 +1,60 @@
+package com.tp.Nile.controllers;
+
+import com.tp.Nile.exceptions.InvalidSpecIdException;
+import com.tp.Nile.exceptions.NullSpecIdException;
+import com.tp.Nile.models.Specification;
+import com.tp.Nile.services.SpecificationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RequestMapping("/api")
```

Comment:
Could you refactor the request mapping to be /api/specs and then each route can build off that. I know this seems kind of OCD, but it makes it easier to remember the routes when making requests.